cmd/cluster-agent/api/server_test.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateTokenMiddleware(t *testing.T) {
+	config.Datadog.Set("cluster_agent.auth_token", "abc123")
+	util.SetDCAAuthToken()
+
+	tests := []struct {
+		path               string
+		expectedStatusCode int
```

Comment:
What do you think about adding the token in the test to test an additional case where the token is different and thus the query fails?